File: services/userService/src/main/java/com/example/userService/config/WebConfig.java
package com.example.userService.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class WebConfig implements WebMvcConfigurer {

    private final SecurityProperties securityProperties;

    public WebConfig(SecurityProperties securityProperties) {
        this.securityProperties = securityProperties;
    }

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(new ApiClientAuthInterceptor(securityProperties))
                .addPathPatterns("/**");
    }
}

File: services/credit/src/main/kotlin/com/example/credit/config/WebConfig.kt
package com.example.credit.config

import com.example.credit.service.role.RoleAccessService
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebConfig(
    private val securityProperties: SecurityProperties,
    private val roleAccessService: RoleAccessService
) : WebMvcConfigurer {

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(ApiClientAuthInterceptor(securityProperties))
            .addPathPatterns("/**")
        registry.addInterceptor(RoleAccessInterceptor(roleAccessService))
            .addPathPatterns("/client/**", "/employee/**")
    }
}
